### PR TITLE
DCOS-41268: port table docs from google docs

### PR DESCRIPTION
**Pull request by lilysamimi**
Only moved over things that are currently in the ui-kit. Will port over more once we have more table components built out in the ui-kit. 

[Design System Docs](https://docs.google.com/document/d/1HiT0YTZe930DdSukKapahfzMG9Dc-pEKIjFUBuVeyw0/edit#heading=h.earj3o607tlu)

Closes DCOS-41268

### CODE REVIEW

**Comment by GeorgiSTodorov** [comment]
Commit was failing with linting errors:
```
ERROR: /jenkins/workspace/tend_UI-Kit_Pipeline_PR-230-STR22XQRBPQISJT247YO32WNYY6ZG4HANJQNMATWKUPGZZ4PUFYQ/packages/table/components/Table.tsx[22, 3]: Delete `··`

ERROR: /jenkins/workspace/tend_UI-Kit_Pipeline_PR-230-STR22XQRBPQISJT247YO32WNYY6ZG4HANJQNMATWKUPGZZ4PUFYQ/packages/table/components/Table.tsx[23, 1]: asterisks in jsdoc must be aligned

ERROR: /jenkins/workspace/tend_UI-Kit_Pipeline_PR-230-STR22XQRBPQISJT247YO32WNYY6ZG4HANJQNMATWKUPGZZ4PUFYQ/packages/table/components/Table.tsx[24, 1]: asterisks in jsdoc must be aligned
```

I pushed a commit that should fix this.

**Comment by nLight** [comment]
I went ahead and dropped the package-lock changes and squashed @GeorgiSTodorov commit. Please give it another shot

File: packages/table/README.md [hunk 1]
```diff
@@ -1,19 +1,59 @@
...
+
+1. Primary column: The primary is always the first column in the table. This column should be the identifying attribute of your data set. 
+
+2. Long text: Truncate long text to fit in the row and add an ellipses with a tooltip on hover to show the full text. 
+
+3. Table Headings: These should describe the content in the column. Try to stick to one word. Table heading should be in sentence case. 
+
+4. Empty values: Empty values should be expressed in the table as “N/A”. 
+
+5. Maximum column width: To handle columns that tend to have long text, the maximum column width allowed is 250 pixels.
```
Comment (mperrotti) [inline packages/table/README.md]: When/if the maximum column width changes, this documentation will be out of date. I'd recommend removing this item

File: packages/table/README.md [hunk 2]
```diff
@@ -1,19 +1,59 @@
...
+
+For Detail tables, content should always be left aligned no matter if it’s numeric or string. 
+
+## Sorting
+
+By default one column is sorted in each table in ascending or descending order. It’s up to the design lead to decide which column and what order. Columns can sort both A-Z and 0-100. 
+
+* Upward chevron is for ascending the items in the column. Ascending means smallest to largest, increasing in size.
+* Downward chevron is for descending the items in the column. Descending means largest to smallest, decreasing in size.
+* Chevron should always be placed 6px to the right of the column header and centered with the column header.
```
Comment (mperrotti) [inline packages/table/README.md]: I'd also remove this. The spacing is already baked into the code (and probably the Sketch symbols too) :)

**Comment by GeorgiSTodorov** [comment]
@mperrotti I am not sure if Lily is keeping an eye on this, but I can address your comments if you don't mind so that we can get this merged.

**Comment by lilysamimi** [comment]
Thank you @GeorgiSTodorov! Appreciate it! I definitely was not following this.

**Pull request merged**